File: Soluciones-Angel-Alcantar1/ejercicio4.py
#ejercicio4
#Angel alcántar
#python.

"""
4. Create an online shipping system with the following features:
* 		The system has a login that locks after the third failed attempt.
* 		Display a menu that allows: Sending a package, exiting the system.
* 		To send a package, sender and recipient details are required.
* 		The system assigns a random package number to each sent package.
* 		The system calculates the shipping price. $2 per kg.
* 		The user must input the total weight of their package, and the system should display the amount to pay.


* 		The system should ask if the user wants to perform another operation. If the answer is yes, it should return to the main menu. If it's no, it should close the system.
"""
import random
#datos de login
usuarios = {"admin": "password"}
def login(usuarios):
  intentos = 0
  while(intentos<3):
    usuario = input("Ingrese su usuario: ")
    password = input("Ingrese su contraseña: ")
    if usuarios.get(usuario) == password:
      return True
    else:
      intentos += 1
      print("Usuario o contraseña incorrectos.")
  return False

#función para enviar un paquete
def enviar_paquete():
  #recpilar nombre del remitente, destinatario, peso del paquete, randomizar el número de paquete y calcular a la vez que se muestra el precio.
  nombre_remitente = input("Ingrese el nombre del remitente: ")
  nombre_destinatario = input("Ingrese el nombre del destinatario: ")
  peso = float(input("Ingrese el peso del paquete: "))
  numero_paquete = random.randint(1000, 9999)
  precio = calcular_precio_envio(peso)
  print(f"Paquete #{numero_paquete} enviado. Precio a pagar: ${precio}")
  #función para calcular el precio del envío
def calcular_precio_envio(peso):
  return peso * 2

#función para el menú
def sistema_envio():
    while True:
        if login(usuarios):
            while True:  # Añadir un bucle para volver al menú después de cada operación
                print("1. Enviar paquete")
                print("2. Salir")
                opcion = input("Ingrese una opción: ")
                if opcion == "1":
                    enviar_paquete()
                elif opcion == "2":
                    print("Gracias por usar nuestro sistema de envío.")
                    break
                    exit()
                else:
                    print("Opción no válida.")

                otra_operacion = input("¿Desea realizar otra operación? (sí/no): ").lower()
                if otra_operacion != "sí":
                    break
                
        else:
            print("Ha excedido el número de intentos de inicio de sesión.")
            break

if __name__ == "__main__":
  sistema_envio()

File: Exercises_done/Ejercicio4.java
package programmingBlind.Level0;

import java.util.Scanner;

import java.util.Random;

public class Ejercicio4 {

	private static final String NOMBREUSUARIO = "Garlopa";
	private static final String PASSWORD = "Garlopa";
	private static final int CANTMAXINTENTOSLOGIN = 3;
	private static final double COSTOCARGOPORKILO = 2;
	private static int intentosLogin = 0;
	private static String optNuevaOperacion;
	private static double pesoPaquete;
	private static Scanner scanner = new Scanner(System.in);

	public static void main(String[] args) {
		login();

		boolean salidaDelSistema = false;

		do {
			System.out.println("Bienvenido al sistema de envio de paquetes.");
			System.out.println("Por favor elija la opción de lo que desea hacer:");
			System.out.println("[1] - Para enviar un paquete.");
			System.out.println("[2] - Para salir del sistema");

			int eleccion = scanner.nextInt();

			switch (eleccion) {

			case 1:

				enviarPaquete();
				break;

			case 2:

				salidaDelSistema = true;
				break;

			default:

				System.out.println("Opcion no válida, por favor intente nuevamente");

			}

			if (!salidaDelSistema) {

				System.out.println("Desea realizar una nueva operación? [S]I / [N]O");
				optNuevaOperacion = scanner.next();

				if (!optNuevaOperacion.equalsIgnoreCase("s")) {

					salidaDelSistema = true;

				}

			}

		} while (!salidaDelSistema);

		System.out.println("Gracias por enviar paquetes a través de nosotros, que tenga buen dia!!!");

	}

	private static void login() {

		do {

			System.out.println("Por favor ingrese usuario y contraseña.");

			System.out.print("Usuario: ");
			String nombreUsuario = scanner.next();

			System.out.print("Contraseña: ");
			String contraseña = scanner.next();

			if (nombreUsuario.equals(NOMBREUSUARIO) && contraseña.equals(PASSWORD)) {

				System.out.println("Login realizado con exito...");
				break;
			} else {

				System.out.println("Nombre de usuario y/o contraseña incorrectos, por favor intente nuevamente");
				intentosLogin++;

				if (intentosLogin == CANTMAXINTENTOSLOGIN) {

					System.out.println("Se alcanzó la cantidad máxima de intentos, salida del sistema");
					System.exit(0);
				}
			}

		} while (true);

	}

	private static void enviarPaquete() {

		double costoPaquete;
		String remitente;
		String destinatario;

		System.out.println("Por favor, ingrese el peso en KG del paquete a enviar: ");
		pesoPaquete = scanner.nextDouble();

		costoPaquete = pesoPaquete * COSTOCARGOPORKILO;
		System.out.println("El costo por envíar el paquete será de: " + costoPaquete);

		System.out.println("Ingrese el nombre del remitente");
		remitente = scanner.next();

		System.out.println("Ingrese el nombre del destinatario");
		destinatario = scanner.next();

		System.out.println("Datos necesarios completados con exito. Detalles del paquete a enviar:");
		System.out.println("Se enviará un paquete nro. " + generarNumeroPaquete());
		System.out.println("Cuyo Remitente es: " + remitente);
		System.out.println("El Destinatario será: " + destinatario);
		System.out.println("El costo de envio del mismo será: " + costoPaquete);

	}

	private static int generarNumeroPaquete() {

		Random random = new Random();

		int numeroPaquete = random.nextInt(1000) + 1;

		return numeroPaquete;

	}

}
